### PR TITLE
Add examples for Chapter 20

### PR DESCRIPTION
**Pull request by Peter-VanWeert**
Exercise solutions will follow later...

### CODE REVIEW

File: Examples/NoModules/Chapter 20/Ex20_11/Ex20_11.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    return std::abs(x - number_to_search_for) < std::abs(y - number_to_search_for);
+  }};
+
+  std::cout << "The number nearest to " << number_to_search_for << " is "
+            << *std::min_element(begin(numbers), end(numbers), nearer) << std::endl;
+  std::cout << "The number furthest from " << number_to_search_for << " is "
+            << *std::max_element(begin(numbers), end(numbers), nearer) << std::endl;
+
+/*
+  const auto [nearest, furthest] =
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_11/Ex20_11.cpp:29]: uniform init

File: Examples/NoModules/Chapter 20/Ex20_11A/Ex20_11A.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    return std::abs(x - number_to_search_for) < std::abs(y - number_to_search_for);
+  }};
+
+  std::cout << "The number nearest to " << number_to_search_for << " is "
+            << *std::ranges::min_element(numbers, nearer) << std::endl;
+  std::cout << "The number furthest from " << number_to_search_for << " is "
+            << *std::ranges::max_element(numbers, nearer) << std::endl;
+
+/*
+  const auto [nearest, furthest] =
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_11A/Ex20_11A.cpp:30]: uniform init

File: Examples/NoModules/Chapter 20/Ex20_12/Ex20_12.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#include <vector>
+#include <algorithm>
+#include "Box.h"      // From Ex13_03A
+
+int main()
+{
+  std::vector boxes{ Box{1,2,3}, Box{5,2,3}, Box{9,2,1}, Box{3,2,1} };
+
+  // Define a lambda functor to print the result of find() or find_if():
+  auto print_result = [&boxes] (auto result)
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_12/Ex20_12.cpp:12]: uniform init

File: Examples/NoModules/Chapter 20/Ex20_12/Ex20_12.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                << (result - begin(boxes)) << std::endl;
+  };
+
+  // Find an exact box
+  Box box_to_find{ 3,2,1 };
+  auto result{ std::find(begin(boxes), end(boxes), box_to_find) };
+  print_result(result);
+
+  // Find a box with a volume larger than that of box_to_find
+  const auto required_volume = box_to_find.volume();
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_12/Ex20_12.cpp:27]: uniform init

File: Examples/NoModules/Chapter 20/Ex20_12A/Ex20_12A.cpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                << (result - begin(boxes)) << std::endl;
+  };
+
+  // Find an exact box
+  Box box_to_find{ 3,2,1 };
+  auto result{ std::ranges::find(boxes, box_to_find) };
+  print_result(result);
+
+  // Find a box with a volume larger than that of box_to_find
+  const auto required_volume = box_to_find.volume();
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_12A/Ex20_12A.cpp:27]: uniform init

File: Examples/NoModules/Chapter 20/Ex20_12A/Ex20_12A.cpp [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+#include <vector>
+#include <algorithm>
+#include "Box.h"      // From Ex13_03A
+
+int main()
+{
+  std::vector boxes{ Box{1,2,3}, Box{5,2,3}, Box{9,2,1}, Box{3,2,1} };
+
+  // Define a lambda functor to print the result of find() or find_if():
+  auto print_result = [&boxes] (auto result)
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_12A/Ex20_12A.cpp:12]: uniform init

File: Examples/NoModules/Chapter 20/Ex20_13/Ex20_13.cpp [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+#include <algorithm>
+
+std::set<int> fillSet_1toN(size_t N);        // Fill a set with 1, 2, ..., N
+void printVector(const std::vector<int>& v); // Print the contents of a vector to std::cout
+
+int main()
+{
+  const size_t num_numbers{20};
+
+  const auto numbers = fillSet_1toN(num_numbers);
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_13/Ex20_13.cpp:14]: uniform init

File: Examples/NoModules/Chapter 20/Ex20_13/Ex20_13.cpp [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  std::vector<int> odd_numbers( numbers.size() ); // Caution: not { numbers.size() } here!
+  auto end_odd_numbers = std::copy_if(begin(numbers), end(numbers), begin(odd_numbers),
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_13/Ex20_13.cpp:17]: ****

File: Examples/NoModules/Chapter 20/Ex20_13A/Ex20_13A.cpp [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+#include <iterator>	           // For std::back_inserter()
+
+std::set<int> fillSet_1toN(size_t N);        // Fill a set with 1, 2, ..., N
+void printVector(const std::vector<int>& v); // Print the contents of a vector to std::cout
+
+int main()
+{
+  const size_t num_numbers{20};
+
+  const auto numbers = fillSet_1toN(num_numbers);
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_13A/Ex20_13A.cpp:15]: uniform init

File: Examples/NoModules/Chapter 20/Ex20_13B/Ex20_13B.cpp [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+#include <iterator>	           // For std::back_inserter()
+
+std::set<int> fillSet_1toN(size_t N);        // Fill a set with 1, 2, ..., N
+void printVector(const std::vector<int>& v); // Print the contents of a vector to std::cout
+
+int main()
+{
+  const size_t num_numbers{ 20 };
+
+  const auto numbers = fillSet_1toN(num_numbers);
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_13B/Ex20_13B.cpp:16]: uniform init

File: Examples/NoModules/Chapter 20/Ex20_16/Ex20_16.cpp [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+#include <algorithm>
+#include <iterator>   // For std::back_inserter()
+#include <ranges>
+#include "Box.h"
+
+using namespace std::ranges::views;
+
+int main()
+{
+  // Our challenge: give a sequence of boxes,
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_16/Ex20_16.cpp:13]: "given"

File: Examples/NoModules/Chapter 20/Ex20_17/Ex20_17.cpp [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+#include <ranges>   // For views, range factories, and range adaptors
+
+using namespace std::ranges::views;
+
+bool isEven(int i) { return i % 2 == 0; }
+int squared(int i) { return i * i; }
+
+int main()
+{
+  for (int i : iota(1, 10))       // Lazily generate range [1,10)
```
Comment (MarcGregoire) [inline Examples/NoModules/Chapter 20/Ex20_17/Ex20_17.cpp:12]: Can't verify with VC++, doesn't support iota yet

**Comment by MarcGregoire** [comment]
Seems like the exercise solutions are missing?

**Review by MarcGregoire — CHANGES_REQUESTED** [review]

File: Examples/NoModules/Chapter 20/Ex20_05/Ex20_05.cpp [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+  WordCounts wordCounts = countWords(words);
+  showWordCounts(wordCounts);
+}
+
+Words extractWords(std::string_view text, std::string_view separators)
+{
+  Words words;
+  size_t start{ text.find_first_not_of(separators) };    // Start 1st word
+  size_t end{};                                          // Index for the end of a word
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 20/Ex20_05/Ex20_05.cpp:42]: True. Need not be. But it's like that in the book, no doubt, so I'll leave it.

File: Examples/NoModules/Chapter 20/Ex20_17/Ex20_17.cpp (hunk 12 above)
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 20/Ex20_17/Ex20_17.cpp:12]: Tip: you can use the ranges-v3 library.

**Pull request merged**